massa-api/src/api_trait.rs

```diff
@@ -38,7 +38,7 @@ pub trait MassaApi {
     #[subscription(
         name = "subscribe_new_blocks_headers" => "new_blocks_headers",
         unsubscribe = "unsubscribe_new_blocks_headers",
-        item = BlockHeader
+        item = SecureShare<BlockHeader, BlockId>
```

Comment:
IMHO, the correct naming is SignedXXX

We accept serialized_data in SendXXXX methods... so this should be kept and we'll add SignedXXX for data output ?

Reply:
We accept the serialized data because it needs to be done by the user sending (the block for example) to make the signature so we also use it to deserialize the block. It's not mandatory to have it on the return of a block as we also have the data already deserialized for me it will just create big object humanly unreadable. (maybe a personal opinion)

Reply:
What you are describing was the initial implementation... may be it's time to rollback with two separate structures to handle this.

Reply:
ok let's keep it like that